src/Compilers/CSharp/Portable/BoundTree/BoundSequencePoint.cs

```diff
@@ -2,22 +2,38 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
+
 using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class BoundSequencePoint
     {
-        public static BoundStatement Create(CSharpSyntaxNode syntax, TextSpan? part, BoundStatement statement, bool hasErrors = false)
+        public static BoundStatement Create(SyntaxNode? syntax, TextSpan? part, BoundStatement statement, bool hasErrors = false)
```

Comment:
Should we add a `new` `Syntax` getter, that hides the base one and is `Syntax?`?

Reply:
No.  Almost every client should be able to assume syntax is not null.

---
In reply to: [387975092](https://github.com/dotnet/roslyn/pull/42068#discussion_r387975092) [](ancestors = 387975092)

Reply:
No, that would prevent the auto-generated Update method from compiling without a warning.

---
In reply to: [387975092](https://github.com/dotnet/roslyn/pull/42068#discussion_r387975092) [](ancestors = 387975092)

Reply:
> No, that would prevent the auto-generated Update method from compiling without a warning.

Then perhaps we should add a way to add a suppression to the generated `Update` method. Can you file a bug to track? If we're lying about what `Syntax` can return, we're likely going to get bitten by it at some point.

Reply:
Filed https://github.com/dotnet/roslyn/issues/42389